bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/discovery/GreeDeviceFinder.java

```diff
@@ -102,9 +102,18 @@ public void scan(DatagramSocket clientSocket, String broadcastAddress, boolean s
                         continue;
                     }
 
+                    String decryptedMsg;
                     // Decrypt message - a GreeException is thrown when something went wrong
-                    String decryptedMsg = scanResponseGson.decryptedPack = GreeCryptoUtil
-                            .decryptPack(GreeCryptoUtil.getAESGeneralKeyByteArray(), scanResponseGson.pack);
+                    // If there is a tag property in the responce then use GCM decryption
```

Comment:
could you move the logic to GreeCryptoUtil class, this layer here should not deal with it, just call a encrypt/decrypt method doing the details